src/components/views/settings/JoinRuleSettings.tsx

```diff
@@ -89,6 +97,11 @@ const JoinRuleSettings = ({ room, promptUpgrade, aliasWarning, onError, beforeCh
         label: _t("Private (invite only)"),
         description: _t("Only invited people can join."),
         checked: joinRule === JoinRule.Invite || (joinRule === JoinRule.Restricted && !restrictedAllowRoomIds?.length),
+    }, {
+        value: JoinRule.Knock,
+        label: _t("Ask to join"),
+        description: _t("Requires users to be granted access in order to join"),
+        checked: joinRule === JoinRule.Knock && knockingEnabled === true,
```

Comment:
I think we might want to show this even if `knockingEnabled === false` to avoid lying to the user

Reply:
This variable is to check whether `knocking` option is enabled in the labs flag or not, if not enabled will it be right to show this option?

Reply:
Yes, though even if it is disabled, the room's join rule could be `JoinRule.Knock` - we should be smarter and show the user what the actual join rule is no matter what but we shouldn't allow the user to set the join rule to `JoinRule.Knock` if `knockingEnabled === false`

Reply:
it's sort of expected that the labs flag will be moderately short-lived, so would be okay with it just being wrong when people don't have the labs flag enabled.